### PR TITLE
Keep nvcc for MSDeepSpeed Docker image

### LINKED ISSUE #1710: [docker] DeepSpeed image should contain nvcc
Currently, "pytorchignite/msdp-apex:latest" docker image can not run cifar10 DeepSpeed example failing with error:
```
...
    basic_optimizer = self._configure_basic_optimizer(model_parameters)                                                                                                                          
  File "/opt/conda/lib/python3.8/site-packages/deepspeed/runtime/engine.py", line 661, in _configure_basic_optimizer
    optimizer = FusedAdam(model_parameters, **optimizer_parameters)
  File "/opt/conda/lib/python3.8/site-packages/deepspeed/ops/adam/fused_adam.py", line 72, in __init__
    fused_adam_cuda = FusedAdamBuilder().load()
  File "/opt/conda/lib/python3.8/site-packages/deepspeed/ops/op_builder/builder.py", line 174, in load
    return self.jit_load(verbose)
...
  File "/opt/conda/lib/python3.8/subprocess.py", line 1702, in _execute_child         
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/cuda/bin/nvcc'
    raise child_exception_type(errno_num, err_msg, err_filename)
```

### PR DESCRIPTION
**Pull request by trsvchn**
Fixes #1710 

Description:

Prevents `nvcc` from being removed by excluding `g++` removing step from Docker file

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by trsvchn** [comment]
@vfdev-5 BTW, should we do the same for other images?

**Comment by sdesrozis** [comment]
@trsvchn thank you. What is the size of the image keeping g++ ?

**Comment by trsvchn** [comment]
@sdesrozis well... it's

```bash
pytorchignite/msdp-apex          latest                        ...   ...    14.6GB
```

File: docker/msdp/Dockerfile.msdp-apex [hunk 1]
```diff
@@ -67,10 +67,7 @@ RUN pip install --upgrade --no-cache-dir pytorch-ignite \
 # replace pillow with pillow-simd
 RUN apt-get update && apt-get -y install --no-install-recommends g++ && \
```
Comment (vfdev-5) [inline docker/msdp/Dockerfile.msdp-apex:68]: It is even unnecessary to install it as it is already present on the image.

**Comment by vfdev-5** [comment]
I also rebuilt it and have 
```
pytorch/pytorch                 1.7.1-cuda11.0-cudnn8-devel       7554ac65eba5        5 weeks ago         12.9GB
pytorchignite/msdp-apex         test                              154f299788d9        23 hours ago        14.5GB
```

**Review by vfdev-5 — APPROVED** [review]
Thanks @trsvchn !
I'll merge it like that, but we have to fix our CI pipeline for building docker images

**Pull request merged**